internal/wait/wait.go

```diff
@@ -41,6 +42,34 @@ func ClusterPolicyReady(apiClient *clients.Settings, clusterPolicyName string, p
 		})
 }
 
+// NicClusterPolicyReady Waits until nicClusterPolicy is Ready.
+func NicClusterPolicyReady(apiClient *clients.Settings, nicClusterPolicyName string, pollInterval,
+	timeout time.Duration) error {
+	return wait.PollUntilContextTimeout(
+		context.TODO(), pollInterval, timeout, true, func(ctx context.Context) (bool, error) {
+			nicClusterPolicy, err := nvidianetwork.Pull(apiClient, nicClusterPolicyName)
+
+			if err != nil {
+				glog.V(gpuparams.GpuLogLevel).Infof("ClusterPolicy pull from cluster error: %s\n", err)
+
+				return false, err
+			}
+
+			if nicClusterPolicy.Object.Status.State == "ready" {
+				glog.V(gpuparams.GpuLogLevel).Infof("NicClusterPolicy %s in now in %s state",
+					nicClusterPolicy.Object.Name, nicClusterPolicy.Object.Status.State)
+
+				// this exists out of the wait.PollImmediate()
+				return true, nil
+			}
+
+			glog.V(gpuparams.GpuLogLevel).Infof("NicClusterPolicy %s in now in %s state",
+				nicClusterPolicy.Object.Name, nicClusterPolicy.Object.Status.State)
+
```

Comment:
This code can be simplified to print the status always (which it already does), then return depending on the status. E.g.
```go
			glog.V(gpuparams.GpuLogLevel).Infof("NicClusterPolicy %s in now in %s state",
				nicClusterPolicy.Object.Name, nicClusterPolicy.Object.Status.State)
			return nicClusterPolicy.Object.Status.State == networkoperator.StateReady, nil
```

Reply:
Good suggestion ! Done !